### PR TITLE
Feature/cax 1 prs@183 Additional Integration Tests for Query API

### PR DESCRIPTION
**Pull request by cpeeyush**
- Additional Integration Tests for current Query API implementation.

### CODE REVIEW

File: coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/test/GetPartsTreeByObjectIdIntegrationTests.java [hunk 1]
```diff
@@ -97,12 +101,107 @@ public void getPartsTreeByObjectId_notExistingOneId_returns404() throws Exceptio
...
+                        .pathParam(ONE_ID_MANUFACTURER, PART_ONE_ID)
+                        .pathParam(OBJECT_ID_MANUFACTURER, PART_OBJECT_ID)
+                        .queryParam(VIEW, AS_MAINTAINED)
+                        .queryParam(ASPECT, "CE")
+                .when()
+                        .get(PATH)
+                .then()
+                        .assertThat()
+                        .statusCode(HttpStatus.OK.value())
+                        .extract().asString();
```
Comment (marcgs) [inline coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/test/GetPartsTreeByObjectIdIntegrationTests.java:180]: We already checked the full response in previous tests, thus in this one I would just go with checking that the CE aspect is in the response using [jsonPath](https://github.com/json-path/JsonPath)

File: coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/test/GetPartsTreeByObjectIdIntegrationTests.java [hunk 2]
```diff
@@ -97,12 +101,107 @@ public void getPartsTreeByObjectId_notExistingOneId_returns404() throws Exceptio
...
+                given()
+                        .pathParam(ONE_ID_MANUFACTURER, "BOSCH")
+                        .pathParam(OBJECT_ID_MANUFACTURER, "CHOQAST")
+                        .queryParam(VIEW, AS_MAINTAINED)
+                .when()
+                        .get(PATH)
+                .then()
+                        .assertThat()
+                        .statusCode(HttpStatus.OK.value())
+                        .extract().asString();
```
[the remark this answers is not in the document]
Comment (cpeeyush) [reply coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/test/GetPartsTreeByObjectIdIntegrationTests.java:197]: Updated. and removed the empty response file. Thanks

File: coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/test/GetPartsTreeByObjectIdIntegrationTests.java [hunk 3]
```diff
@@ -53,56 +59,144 @@ public void getPartsTreeByObjectId_maintainedView_success() throws Exception {
...
+    @Test
+    public void getPartsTreeByObjectId_notExistingOneId_emptyResponse() throws Exception {
+        var response =
+            given()
+                .pathParam(ONE_ID_MANUFACTURER, "not-existing-one-id")
+                .pathParam(OBJECT_ID_MANUFACTURER, PART_OBJECT_ID)
+                .queryParam(VIEW, AS_MAINTAINED)
+            .when()
+                .get(PATH)
+            .then()
```
Comment (marcgs) [inline coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/test/GetPartsTreeByObjectIdIntegrationTests.java:90]: rest assured supports:
```
.then()
.assertThat()
.statusCode(HttpStatus.OK.value())
.body("relationships", isEmpty())

```

**Review by marcgs — APPROVED** [review]

**Comment by marcgs** [comment]
@cpeeyush I have added coverage for the AS_BUILT view as well (returning the same as AS_MAINTAINED)

File: coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/test/GetPartsTreeByVinIntegrationTests.java [hunk 4]
```diff
@@ -62,7 +66,7 @@ public void getPartsTreeByVin_notExistingVIN_returns404() {
     @Test
     public void getPartsTreeByVin_noView_returns400() {
         given()
-            .pathParam("vin", VIN)
+            .pathParam(VIN, SAMPLE_VIN)
         .when()
             .get(PATH)
         .then()
```
Comment (chrislomonico) [inline coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/test/GetPartsTreeByVinIntegrationTests.java:72]: should this also assert for a bad request?

File: coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/test/GetPartsTreeByVinIntegrationTests.java [hunk 5]
```diff
@@ -50,8 +54,8 @@ public void getPartsTreeByVin() throws Exception {
     @Test
     public void getPartsTreeByVin_notExistingVIN_returns404() {
         given()
-            .pathParam("vin", "not-existing-vin")
-            .queryParam("view", AS_MAINTAINED)
+            .pathParam(VIN, "not-existing-vin")
+            .queryParam(VIEW, AS_MAINTAINED)
         .when()
             .get(PATH)
         .then()
```
Comment (chrislomonico) [inline coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/test/GetPartsTreeByVinIntegrationTests.java:61]: should this as well be asserting the response? (applies to all where _returnsXXX is implied with the name

**Review by chrislomonico — CHANGES_REQUESTED** [review]
see comments regarding the tests that imply an error response

File: coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/test/GetPartsTreeByVinIntegrationTests.java (hunk 5 above)
Comment (cpeeyush) [reply coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/test/GetPartsTreeByVinIntegrationTests.java:61]: Do you mean where we assert response body as well ? We have a follow up task for that to add an error response body and will also cover the assertion by updating our tests. Right now we only have response status codes from API.

**Review by chrislomonico — APPROVED** [review]

**Pull request merged**